src/wp-includes/general-template.php

```diff
@@ -3326,9 +3326,10 @@ function feed_links_extra( $args = array() ) {
 	 */
 	$args = apply_filters( 'feed_links_extra_args', $args );
 
-	if ( is_singular() ) {
-		$id   = 0;
-		$post = get_post( $id );
+	// Singular objects only, excluding a single `show_on_front` Home Page.
+	$queried_object = get_queried_object();
+	if ( is_singular() && ! is_home() && $queried_object instanceof WP_Post ) {
```

Comment:
I think I'll actually remove this change because I just tested and `is_singular()` and `is_home()` are mutually exclusive. Even when `$wp_query->is_posts_page` is true, the `is_singular()` check remains false.

```suggestion
	$queried_object = get_queried_object();
	if ( is_singular() && $queried_object instanceof WP_Post ) {
```

In any case, the use of `$queried_object instanceof WP_Post` will still solve the original problem.